src/application/selector/xml/XmlSelectorAdapter.js

```diff
@@ -0,0 +1,29 @@
+const xpath = require('xpath');
+const { XMLSerializer, DOMParser } = require('xmldom');
+const DataSelectorPort = require('../../../domain/common/port/DataSelectorPort');
+const XmlResultFormatter = require('./XmlResultFormatter');
+
+
+class XmlSelectorAdapter extends DataSelectorPort {
+  constructor() {
+    super();
+    this.domParser = new DOMParser();
+    this.xmlSerializer = new XMLSerializer();
+  }
+
+  canHandle(contentType) {
+    return ['xml', 'html'].includes(contentType);
+  }
+
+  select(data, path) {
+    const doc = this.domParser.parseFromString(data);
+    this.xmlSerializer.serializeToString(doc); // this throw an error when doc is invalid xml
+
+    const expression = path === null ? '//*' : path;
```

Comment:
Why call XPath with '//*' ? Can't we just return doc in such case? If we can avoid calling XPath there lets go for it - it introduces risk and it's also costly.

If we need to call XPath, I'm not so sure that '//*' will select every XML Node type in the document. Perhaps it's worth to write a test for this case - one with a document that contains many different XML Node types.